atlasdb-coordination-impl/src/main/java/com/palantir/atlasdb/coordination/SequenceAndBound.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.coordination;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+/**
+ * A pair of a sequence number and a bound on the validity of the value associated with that sequence number.
+ */
+@Value.Immutable
+@JsonSerialize(as = ImmutableSequenceAndBound.class)
+@JsonDeserialize(as = ImmutableSequenceAndBound.class)
```

Comment:
IIRC default is fail on unknown properties, which I _think_ is fine for the usage here since it's just which entry is the relevant one + the validity bound. We'll want ignore unknown on the entry object.